src/gen/opcodes.rs

```diff
@@ -60,53 +60,74 @@ pub const SOFTFORK: ConditionOpcode = 90;
 pub const CREATE_COIN_COST: Cost = 1800000;
 pub const AGG_SIG_COST: Cost = 1200000;
 
+// when ENABLE_SOFTFORK_CONDITION is enabled
+// 2-byte condition opcodes whose first byte is 1 have costs, despite being unknown:
+
+// cost = 1000000 + (byte[1] & 0x7) * 200000
+
+pub fn compute_unknown_condition_cost(op: ConditionOpcode) -> Cost {
+    match op {
+        256..=511 => 1000000 + ((op & 0x7) as Cost) * 200000,
+        _ => 0,
+    }
+}
+
 pub fn parse_opcode(a: &Allocator, op: NodePtr, flags: u32) -> Option<ConditionOpcode> {
     let buf = match a.sexp(op) {
         SExp::Atom(_) => a.atom(op),
         _ => return None,
     };
-    if buf.len() != 1 {
-        return None;
-    }
-
-    match buf[0] {
-        AGG_SIG_UNSAFE
-        | AGG_SIG_ME
-        | CREATE_COIN
-        | RESERVE_FEE
-        | CREATE_COIN_ANNOUNCEMENT
-        | ASSERT_COIN_ANNOUNCEMENT
-        | CREATE_PUZZLE_ANNOUNCEMENT
-        | ASSERT_PUZZLE_ANNOUNCEMENT
-        | ASSERT_MY_COIN_ID
-        | ASSERT_MY_PARENT_ID
-        | ASSERT_MY_PUZZLEHASH
-        | ASSERT_MY_AMOUNT
-        | ASSERT_SECONDS_RELATIVE
-        | ASSERT_SECONDS_ABSOLUTE
-        | ASSERT_HEIGHT_RELATIVE
-        | ASSERT_HEIGHT_ABSOLUTE
-        | REMARK => Some(buf[0]),
-        _ => {
-            if (flags & ENABLE_SOFTFORK_CONDITION) != 0 && buf[0] == SOFTFORK {
-                Some(buf[0])
-            } else if (flags & ENABLE_ASSERT_BEFORE) != 0 {
-                match buf[0] {
-                    ASSERT_BEFORE_SECONDS_RELATIVE
-                    | ASSERT_BEFORE_SECONDS_ABSOLUTE
-                    | ASSERT_BEFORE_HEIGHT_RELATIVE
-                    | ASSERT_BEFORE_HEIGHT_ABSOLUTE
-                    | ASSERT_CONCURRENT_SPEND
-                    | ASSERT_CONCURRENT_PUZZLE
-                    | ASSERT_MY_BIRTH_SECONDS
-                    | ASSERT_MY_BIRTH_HEIGHT
-                    | ASSERT_EPHEMERAL => Some(buf[0]),
-                    _ => None,
+    if buf.len() == 2 && (flags & ENABLE_SOFTFORK_CONDITION) != 0 {
+        let b0 = buf[0];
+        if b0 != 1 {
+            None
```

Comment:
@richardkiss in case the 2-byte condition code has a redundant leading zero, it's caught here and not interpreted as the 1-byte condition code. For codes > 2 bytes, they are caught on line 130, and also not interpreted as a code whose canonical form has fewer bytes